js/jsonparser.js

```diff
@@ -26,6 +28,7 @@ function updateVertretungsplan() {
           //plan has to be set visible before calling updatePlan() because
           //calculating the correct offset for the plan wouldn't work otherwise
           showById('plan');
+          json = responseJSON;
```

Comment:
Die json Variable wird in der Methode getInfoText() verwendet. Sie ist global, damit der Methode nicht mehr das JSON übergeben werden muss. So kann die Methode in buttonController.js verwendet werden, ohne den Plan erneut herunterladen zu müssen.
Ich würde getInfoText() wieder zu updateInfo(plan) ändern und direkt dort den EventListener auf den Button adden, um die globale Variable zu vermeiden. Dann findet man den EventListener aber eventuell nicht so einfach wieder. Ist das eine bessere Lösung?